langchain/tools/base.py

```diff
@@ -24,6 +89,33 @@ class Config:
         extra = Extra.forbid
         arbitrary_types_allowed = True
 
+    @property
+    def args(self) -> Type[BaseModel]:
+        """Generate an input pydantic model."""
+        if self.args_schema is not None:
+            return self.args_schema
+        return create_args_schema_model_from_signature(self._run)
+
+    def _parse_input(
+        self,
+        tool_input: Union[str, Dict],
+    ) -> BaseModel:
+        """Convert tool input to pydantic model."""
+        pydantic_input_type = self.args
+        if isinstance(tool_input, str):
+            # For backwards compatibility, a tool that only takes
+            # a single string input will be converted to a dict.
+            # to be validated.
+            field_name = next(iter(pydantic_input_type.__fields__))
+            tool_input = {field_name: tool_input}
+        if pydantic_input_type is not None:
+            return pydantic_input_type.parse_obj(tool_input)
+        else:
+            raise ValueError(
+                f"args_schema required for tool {self.name} in order to"
+                f" accept input of type {type(tool_input)}"
+            )
```

Comment:
we can't hit this right? Because if the `pydantic_input_type` is not defined (i.e. self.args), we `create_args_schema_model_from_signature`?